scripts/eval/eval.py

```diff
@@ -56,6 +56,45 @@ def load_model(model_cfg: DictConfig, tokenizer: PreTrainedTokenizerBase,
     return composer_model
 
 
+def log_analytics_details(mosaicml_logger: MosaicMLLogger,
+                          model_config: DictConfig, tokenizer_name: str,
+                          load_path: Union[str,
+                                           None], icl_tasks: Union[str,
+                                                                   ListConfig],
+                          eval_gauntlet_config: Optional[Union[str,
+                                                               DictConfig]]):
+    metrics = {
+        'llmfoundry/llmfoundry_run_type': 'eval',
+        'llmfoundry/tokenizer_name': tokenizer_name,
+        'llmfoundry/model_name': model_config.get('model_name'),
+    }
+
+    if eval_gauntlet_config is not None:
+        metrics['llmfoundry/gauntlet_configured'] = True
+
+    if isinstance(icl_tasks, str):
+        metrics['llmfoundry/icl_configured'] = True
+    elif len(icl_tasks) > 0:
+        metrics['llmfoundry/icl_configured'] = True
+
+    if model_config.get('vocab_size', None) is not None:
+        metrics['llmfoundry/vocab_size'] = model_config.get('vocab_size')
+    if model_config.get('d_model', None) is not None:
+        metrics['llmfoundry/d_model'] = model_config.get('d_model')
+    if model_config.get('n_heads', None) is not None:
+        metrics['llmfoundry/n_heads'] = model_config.get('n_heads')
+
+    if load_path is not None:
+        metrics['llmfoundry/cloud_provider_loading'] = load_path.split(':')[0]
```

Comment:
This would raise a index error for local paths - I would build a helper (can use for train.py too) that safely parses load_path / save_folder

Also nit: maybe loading -> data?

Reply:
I _think_ that in this case, the `split()` function would just return the whole path for local paths. 
<img width="682" alt="Screenshot 2024-03-11 at 1 03 43 PM" src="https://github.com/mosaicml/llm-foundry/assets/141283633/804f4321-7bba-4ace-b41d-7a035f58e889">

Something that I didn't think about though was the naming of these paths. In this case, should we just put `local` instead of the pathname?